src/components/LocalFileUpload/custom-request-upload.ts

```diff
@@ -56,66 +58,68 @@ export default (
-        payload: { newFile: true },
-        type: CLEAR_SIMULARIUM_FILE,
-    });
+    clearSimulariumFile({ newFile: true });
 
     const files: FileHTML[] = Array.from(selectedFiles) as FileHTML[];
 
-    Promise.all(files.map((file: FileHTML) => file.text())).then(
-        (parsedFiles: string[]) => {
-            try {
-                const simulariumFileIndex = findIndex(files, (file) =>
-                    file.name.includes(".simularium")
+    Promise.all(files.map((file: FileHTML) => file.text()))
+        .then((parsedFiles: string[]) => {
+            const simulariumFileIndex = findIndex(files, (file) =>
+                file.name.includes(".simularium")
+            );
+            if (simulariumFileIndex === -1) {
+                throw new Error(
+                    "Trajectory file was not found; please make sure it has a .simularium extension."
                 );
-                // TODO: handle when user doesn't load a .simularium file
-                // if (simulariumFileIndex === -1) {
-                //    throw new Error("Please upload a .simularium file.")
-                // }
+            }
 
-                const simulariumFile: File = files[simulariumFileIndex];
-                const simulariumFileJson: SimulariumFileFormat = JSON.parse(
-                    parsedFiles[simulariumFileIndex]
-                );
-                const geoAssets: { [name: string]: string } = files.reduce(
-                    (acc: { [name: string]: string }, cur, index) => {
-                        if (index !== simulariumFileIndex) {
-                            acc[cur.name] = parsedFiles[index];
-                        }
-                        return acc;
+            const simulariumFile: File = files[simulariumFileIndex];
+            const simulariumFileJson: SimulariumFileFormat = JSON.parse(
+                parsedFiles[simulariumFileIndex]
+            );
+            const geoAssets: { [name: string]: string } = files.reduce(
+                (acc: { [name: string]: string }, cur, index) => {
+                    if (index !== simulariumFileIndex) {
+                        acc[cur.name] = parsedFiles[index];
+                    }
+                    return acc;
+                },
+                {}
+            );
+
+            loadFunction({
+                lastModified: simulariumFile.lastModified,
+                name: simulariumFile.name,
+                data: simulariumFileJson,
+                geoAssets: geoAssets,
+            });
+            // TS thinks onSuccess might be undefined
+            if (onSuccess) {
+                onSuccess(
+                    {
+                        name: simulariumFile.name,
+                        status: "done",
+                        url: "",
                     },
-                    {}
+                    new XMLHttpRequest() // onSuccess needs an XMLHttpRequest arg
                 );
+            }
+        })
+        .catch((error) => {
```

Comment:
just so i understand, this catch happens if any of the files in the `map` call throws or rejects from file.text() ?

Reply:
Yup that's right! The `DOMException` in line 110 for example is thrown from `file.text()` and the error thrown in line 72 is also caught here